jsonapi/marshal.go

```diff
@@ -213,7 +234,7 @@ func getStructRelationships(relationer MarshalLinkedRelations, information Serve
 		relationships[name] = Relationship{}
 		// if referenceType is plural, we need to use an array for data, otherwise it's just an object
 		container := RelationshipDataContainer{}
-		if Pluralize(name) == name {
+		if isToMany(referenceIDs[0].Relationship, referenceIDs[0].Name) {
```

Comment:
`referencedIDs` has to have at least 1 element since `sortedResults` is populated [here](https://github.com/jlburkhead/api2go/blob/75f0e601dcdbfd113d0c6a8c29b6cba3f0b7f480/jsonapi/marshal.go#L222). `referenceIDs[0]` is also accessed [here](https://github.com/jlburkhead/api2go/blob/75f0e601dcdbfd113d0c6a8c29b6cba3f0b7f480/jsonapi/marshal.go#L248) without checking length